// @typescript-eslint/.scope-manager-bFa6QnZF/dist/lib/dom.js
"use strict";
// THIS CODE WAS AUTOMATICALLY GENERATED
// DO NOT EDIT THIS CODE BY HAND
// YOU CAN REGENERATE IT USING yarn generate:lib
Object.defineProperty(exports, "__esModule", { value: true });
exports.dom = void 0;
exports.dom = {
    Account: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'Account',
    },
    AddEventListenerOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AddEventListenerOptions',
    },
    AesCbcParams: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AesCbcParams',
    },
    AesCtrParams: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AesCtrParams',
    },
    AesDerivedKeyParams: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AesDerivedKeyParams',
    },
    AesGcmParams: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AesGcmParams',
    },
    AesKeyAlgorithm: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AesKeyAlgorithm',
    },
    AesKeyGenParams: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AesKeyGenParams',
    },
    Algorithm: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'Algorithm',
    },
    AnalyserOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AnalyserOptions',
    },
    AnimationEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AnimationEventInit',
    },
    AnimationPlaybackEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AnimationPlaybackEventInit',
    },
    AssertionOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AssertionOptions',
    },
    AssignedNodesOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AssignedNodesOptions',
    },
    AudioBufferOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AudioBufferOptions',
    },
    AudioBufferSourceOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AudioBufferSourceOptions',
    },
    AudioContextInfo: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AudioContextInfo',
    },
    AudioContextOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AudioContextOptions',
    },
    AudioNodeOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AudioNodeOptions',
    },
    AudioParamDescriptor: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AudioParamDescriptor',
    },
    AudioProcessingEventInit: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AudioProcessingEventInit',
    },
    AudioTimestamp: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AudioTimestamp',
    },
    AudioWorkletNodeOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AudioWorkletNodeOptions',
    },
    AuthenticationExtensionsClientInputs: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AuthenticationExtensionsClientInputs',
    },
    AuthenticationExtensionsClientOutputs: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AuthenticationExtensionsClientOutputs',
    },
    AuthenticatorSelectionCriteria: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'AuthenticatorSelectionCriteria',
    },
    BiquadFilterOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'BiquadFilterOptions',
    },
    BlobPropertyBag: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'BlobPropertyBag',
    },
    ByteLengthChunk: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ByteLengthChunk',
    },
    CacheQueryOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'CacheQueryOptions',
    },
    CanvasRenderingContext2DSettings: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'CanvasRenderingContext2DSettings',
    },
    ChannelMergerOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ChannelMergerOptions',
    },
    ChannelSplitterOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ChannelSplitterOptions',
    },
    ClientData: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ClientData',
    },
    ClientQueryOptions: {
        eslintImplicitGlobalSetting: 'readonly',
        isTypeVariable: true,
        isValueVariable: false,
        name: 'ClientQueryOptions',
    },
    ClipboardEventInit: {
        eslintImplicitGl
